src/projectworks.ts

```diff
@@ -0,0 +1,83 @@
+import fetch, { Headers, Response } from 'node-fetch';
+import { URLSearchParams } from 'url';
+import type { ProjectWorksLeave, ProjectWorksUser } from './types';
+
+const PW_URL = 'https://api.projectworksapp.com';
+const PW_USERNAME = process.env.PROJECTWORKS_USERNAME;
+const PW_PASSWORD = process.env.PROJECTWORKS_PASSWORD;
+
+const isObjectWithPropName = (ob: unknown, propName: string): boolean => {
+  if (!ob) {
+    return false;
+  }
+
+  if (typeof ob !== 'object') {
+    return false;
+  }
+
+  if (!(propName in ob)) {
+    return false;
+  }
+
+  return true;
+};
+
+const apiGet = async (path: string): Promise<Response> => {
+  const headers = new Headers();
+
+  headers.append('Accept', 'application/json');
+  headers.append(
+    'Authorization',
+    `Basic ${Buffer.from(
+      `${PW_USERNAME as string}:${PW_PASSWORD as string}`
+    ).toString('base64')}`
+  );
+
+  return fetch(`${PW_URL}${path}`, { headers });
+};
+
+const isValidUser = (maybeUser: unknown): maybeUser is ProjectWorksUser => {
+  return isObjectWithPropName(maybeUser, 'UserID');
+};
+
+const isValidLeaves = (
+  maybeLeaves: unknown
+): maybeLeaves is ProjectWorksLeave[] => {
+  if (!Array.isArray(maybeLeaves)) {
+    return false;
+  }
+
+  if (maybeLeaves[0] && !isObjectWithPropName(maybeLeaves[0], 'LeaveID')) {
+    return false;
+  }
+
+  return true;
```

Comment:
This is incorrect, since you're only saying it's not valid leaves if the first item exists exist _and_ it doesn't have a `LeaveID` property - you want this instead:

```suggestion
  if (isObjectWithPropName(maybeLeaves[0], 'LeaveID')) {
    return true;
  }

  return false;
```

Reply:
Ah good catch. I'm not going to accept your suggestion because it would break the pattern of early `false` returns that I've used in other functions in this file but I'll sort the issue.